### PR TITLE
adkinsdk code review 2

### PR DESCRIPTION
**Pull request by derrickadkins**
Analysis of the program.
The IdeaStorm app offers users the ability to generate personalized activities, create an account to save them, reference and check them off a list, add them to a calendar app, and use their Android devices to cure boredom and spark creativity, either individually or in a group setting.

Was the program available in UC Github on time?
Yes

Is the program documented/commented well enough for you to understand?
Yes

Does the program compile?
Yes

Rationale behind your changes.
Kotlin coding conventions

Links to three specific commits that you made to your own group's GitHub repository after Sprint 1's deadline, and before Sprint 2's deadline. 
https://github.com/lh1207/LocoQuest/commit/feaf3d14b777c40a0d7ae97bcfcdd590fc079a2d
https://github.com/lh1207/LocoQuest/commit/faae6d3dd224952ca0cf4e505c1440ee026b97bd
https://github.com/lh1207/LocoQuest/commit/b421ee4e09d7d9f83b854ed11e4ccca6abf06bb0

### CODE REVIEW

**Comment by derrickadkins** [comment]
List three specific technical concepts that you learned from this code review.  Copy and paste samples to demonstrate your point.
1. How to use Composable.
Text(
    text = greeting,
    textAlign = TextAlign.Center,
    style = MaterialTheme.typography.h6
    style = MaterialTheme.typography.h6,
)

2. How to use a lambda as a property.
private val firebaseUser: () -> FirebaseUser? = { FirebaseAuth.getInstance().currentUser }

3. How to properly format a lambda
 user?.let { user ->

File: app/src/main/java/com/ideastorm/v25001/MainViewModel.kt [hunk 1]
```diff
@@ -12,20 +12,19 @@ import com.ideastorm.v25001.service.ActivityService
...
 
-class MainViewModel(var activityService : IActivityService = ActivityService()) : ViewModel() {
-    var activity : MutableLiveData<Activity> = MutableLiveData<Activity>()
-    var user : User? = null
+class MainViewModel(var activityService: IActivityService = ActivityService()) : ViewModel() {
+    var activity: MutableLiveData<Activity> = MutableLiveData<Activity>()
+    var user: User? = null
 
-    private lateinit var firestore : FirebaseFirestore
+    private var firestore: FirebaseFirestore = FirebaseFirestore.getInstance()
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainViewModel.kt:19]: This same suggestion appeared in a previously submitted code review.
When that occurs, only the first gets credit.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 2]
```diff
@@ -40,21 +38,20 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
...
 class MainActivity : ComponentActivity() {
-    var firebaseUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
-    var selectedParticipantOption: String? = null
-    var selectedTypeOption: String? = null
-    var selectedPriceOption: String? = null
-    private val viewModel: MainViewModel by viewModel<MainViewModel>()
+    private val firebaseUser: () -> FirebaseUser? = { FirebaseAuth.getInstance().currentUser }
+    private var selectedParticipantOption: String? = null
+    private var selectedTypeOption: String? = null
+    private var selectedPriceOption: String? = null
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:44]: private modifier appeared in a previously submitted code review.
When that occurs, only the first gets credit.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 3]
```diff
@@ -40,21 +38,20 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
...
         super.onCreate(savedInstanceState)
         setContent {
             viewModel.fetchActivity()
-            firebaseUser?.let {
+            firebaseUser()?.let {
                 val user = User(it.uid, "")
                 viewModel.user = user
             }
 
-            val activity by viewModel.activity.observeAsState(initial = emptyList<Activity>())
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:57]: Why is this removed?

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 4]
```diff
@@ -392,7 +386,7 @@ class MainActivity : ComponentActivity() {
         }
     }
 
-    fun signIn() {
+    private fun signIn() {
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:389]: private access modifier appeared in a previously submitted code review.
When that occurs, only the first gets credit.

File: app/src/main/java/com/ideastorm/v25001/MainViewModel.kt [hunk 5]
```diff
@@ -36,8 +35,7 @@ class MainViewModel(var activityService : IActivityService = ActivityService())
     }
 
     fun saveUser() {
-        user?.let {
-            user ->
+        user?.let { user ->
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainViewModel.kt:38]: Formatting can, and should, be performed automatically, and consistently, by the IDE.  Formatting code is really programmer's preference anyway, and doesn't necessarily decrease technical debt.

**Comment by discospiff** [comment]
From this review:
- Several concepts were covered in previous code reviews. See my commentary there for merge guidance.
- val is generally preferred over var, unless there's a good reason for a variable to be reassigned.  These changes are worth harvesting.
- trailing commas: doesn't necessarily decrease technical debt.
Suggestion: harvest val versus var manually.  For the other suggestions, I'd lean towards merging the previous code review branches that covered these items; no need to merge this one as well.
